chat-android/src/main/java/com/ably/chat/FlowUtils.kt

```diff
@@ -0,0 +1,22 @@
+package com.ably.chat
+
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+/**
+ * Converts a callback-based API to a Flow with an unlimited buffer to handle backpressure
+ */
+internal inline fun <T> transformCallbackAsFlow(crossinline subscribe: ((T) -> Unit) -> Subscription): Flow<T> = flow {
```

Comment:
I am checking edge cases for this.
I had tried snippets for both channels and flows. 
Based on the documentation, I found flows are extensions of channels, also closely tied to lifecycle of a coroutine.

Reply:
I’m not sure I understand what you’re suggesting here.